include/serialization_utils/cereal_cvmat.hpp

```diff
@@ -0,0 +1,92 @@
+/**
+ * @file
+ * @brief Serialization functions for serializing `cv::Mat` with cereal.
+ * @author Patrik Huber
+ * @license Apache-2.0
+ * @todo Move this to some "serialization tools" package
+ *
+ * Taken from
+ * https://www.patrikhuber.ch/blog/2015/05/serialising-opencv-matrices-using-boost-and-cereal/
+ */
+
+// Copyright 2015 Patrik Huber
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// NOTE: The code was modified a bit to fix some compiler errors
+
+#pragma once
+
+#include <cereal/archives/binary.hpp>
```

Comment:
should not we use something like this instead ?
https://github.com/machines-in-motion/shared_memory/blob/master/CMakeLists.txt#L13
and 
https://github.com/machines-in-motion/shared_memory/blob/master/CMakeLists.txt#L18 ?
Possibly it will resulting in clearer compilation errors

Reply:
I'm a bit unsure about this.  Right now it just provides some headers but does not add dependencies to any libraries.  This is convenient as, for example, one does not need to add a dependency on OpenCV when only using the serialization functions for Eigen and vise versa.  I would also assume that if one needs cereal support for Eigen in some package, both cereal and Eigen should already be dependencies there.

But I also see that this could be confusing and considered as a bug of this package, so I'm not sure what's the best solution.

Reply:
I can see the advantage of not having a hard dependency on open cv.
But cereal is used in all header files of the package, so seems to me it makes sense to have it as "hard dependency" (i.e. searched for in the CMakelists.txt) (it would be weird for someone to have serialization_utils in his/her workspace but not having cereal installed, as things would work out only if he/she does not serialize anything, i.e. does not need the serialization_utils package to start with).

Reply:
done